src/Dataset.ts

```diff
@@ -711,15 +718,25 @@ export function add_or_delete_column(
     } else if (data instanceof Uint8Array) {
       tb[field_name] = makeVector({ type: new Uint8(), data, length: data.length }).data[0];
     } else if ((data as Vector<DS.SupportedArrowTypes>).data.length > 0) {
-      tb[field_name] = data.data[0];
+      let newval = data.data[0];
+      if (newval.dictionary) {
+        const dicto = newval as Data<Dictionary<Utf8, Int8 | Int16 | Int32>>;
+        const newv = makeVector({
+          data: dicto.values,  // indexes into the dictionary
+          dictionary: dicto.dictionary as Vector<Utf8>, // keys
+          type: new Dictionary(dicto.type.dictionary, dicto.type.indices, dval++) // increment the identifier.
```

Comment:
I wonder if you could get away with just passing null here, to make it use the internal id generator so you don't conflict again if more than 40k dictionaries get generated in a session https://github.com/apache/arrow/blob/2dcee3f82c6cf54b53a64729fd81840efa583244/js/src/type.ts#L679

Reply:
The issue is that I'm trying to avoid the overhead of generating a new dictionary for every single dictionary, just doing it for ones added to the table... the ids already existing dictionaries can conflict with the internal generator, I think? Prob worth testing because there is a second place I have a number like this.

Another option is to just take the lowest integer is not in the table yet. That's prob the sanest, just a bit more code.